src/librustdoc/clean/auto_trait.rs

```diff
@@ -445,15 +444,15 @@ impl<'a, 'tcx> AutoTraitFinder<'a, 'tcx> {
     fn param_env_to_generics(
         &mut self,
         tcx: TyCtxt<'tcx>,
-        param_env_def_id: DefId,
+        item_def_id: DefId,
```

Comment:
Why was this change made? Did the meaning change, or is `item_def_id` closer to the meaning than `param_env_def_id`? Do `ParamEnv`s even *have* `DefId`s?

Sorry for all the questions; I know almost nothing about `ParamEnv` :)

Reply:
`item_def_id` is closer to the meaning. All items have a ParamEnv, but the same ParamEnv may be used for things other than items (e.g. each expression in the body of a function). Before, there was a semantic mismatch in the representation: it passed in a `Ty`, but not all types are items (for example `&T`). So to get the param env of the type it had to pass in the DefId too. Now it passes in the DefId only, which allows getting both the param env and the type without any additional info.

Reply:
How will `get_blanket_impls` behave now for blanket impls like `impl Foo for &Struct` where the type is `&Struct` but the `type_of` the `item_def_id` is `Struct`? (Unless I'm misunderstanding what the `item_def_id` is referring to.)

Reply:
The same as it did before, all this does is rename the parameter and move the `type_of` into the function. I don't know exactly how it worked before.